src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int NON_ADD_FUNCTION_INDEX = 1;
+    private static final int GROW_SIZE_INDEX = 1;
+    private transient Object[] elementData = new Object[10];
```

Comment:
ArrayList has generic T, which type will be here? Let's use a constructor where we indicate this